imagenet/main.py

```diff
@@ -421,7 +421,10 @@ def update(self, val, n=1):
         self.avg = self.sum / self.count
 
     def all_reduce(self):
-        total = torch.FloatTensor([self.sum, self.count])
+        if (torch.cuda.is_available()):
+            total = torch.cuda.FloatTensor([self.sum, self.count])
+        else:
+            total = torch.FloatTensor([self.sum, self.count])
```

Comment:
Your implementation is correct. But how about using the modern syntax below?

```suggestion
        device = "cuda" if torch.cuda.is_available() else "cpu"
        total = torch.tensor([self.sum, self.count], dtype=torch.float32, device=device)
```

Reply:
Good idea to make use of the modern syntax, its a lot cleaner than using the two different constructors. Have commited the suggestion.